wooting-macro-backend/src/hid_table.rs

```diff
@@ -11,11 +12,61 @@ pub static ref BUTTON_TO_HID: HashMap<Button, u32> = {
         scancode.insert(Button::Left, 0x101);
         scancode.insert(Button::Right, 0x102);
         scancode.insert(Button::Middle, 0x103);
-        scancode.insert(Button::Unknown(4), 0x104);
-        scancode.insert(Button::Unknown(5), 0x105);
+        scancode.insert(Button::Forward, 0x104);
+        scancode.insert(Button::Backward, 0x105);
         scancode
 };}
 
+impl From<&Button> for MouseButton {
+    fn from(value: &Button) -> Self {
+        match *value {
+            Button::Left => MouseButton::Left,
+            Button::Right => MouseButton::Right,
+            Button::Middle => MouseButton::Middle,
+            Button::Forward => MouseButton::Mouse4,
+            Button::Backward => MouseButton::Mouse5,
+            Button::Unknown(_) => MouseButton::Left,
+        }
+    }
+}
+
+impl From<&MouseButton> for Button {
+    fn from(item: &MouseButton) -> Self {
+        match *item {
+            MouseButton::Left => Button::Left,
+            MouseButton::Right => Button::Right,
+            MouseButton::Middle => Button::Middle,
+            MouseButton::Mouse4 => Button::Forward,
+            MouseButton::Mouse5 => Button::Backward,
+        }
+    }
+}
+
+impl From<&MouseButton> for u32 {
+    fn from(value: &MouseButton) -> Self {
+        match *value {
+            MouseButton::Left => 0x101,
+            MouseButton::Right => 0x102,
+            MouseButton::Middle => 0x103,
+            MouseButton::Mouse4 => 0x104,
+            MouseButton::Mouse5 => 0x105,
+        }
+    }
+}
+
+lazy_static! {
+    pub static ref RDEV_MODIFIER_KEYS: [rdev::Key; 8] = [
+        Key::Alt,
+        Key::AltGr,
+        Key::ControlLeft,
+        Key::ControlRight,
+        Key::ShiftLeft,
+        Key::ShiftRight,
+        Key::MetaLeft,
+        Key::MetaRight
+    ];
+}
+
 lazy_static! {
 ///Conversion from HID codes to the library backend enums.
 #[derive(Debug, PartialEq, Hash, std::cmp::Eq)]
```

Comment:
Overall, the changes made here are good. However, this could be a lot cleaner and easier to maintain by using something like [bimap](https://lib.rs/crates/bimap) then you only build up one table and get both directions.

Also, lazy_static isn't needed anymore, you should be able to just do a `once_cell` from std

Reply:
Tried to implement it here, wasn't able to quickly figure out the once_cell yet though: [109c888](https://github.com/WootingKb/wooting-macros/pull/194/commits/109c8883e67e4844e29ee45f305aa2d6547ade38)